### PR TITLE
Added github-merit-badger for OpenSearch

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Coming from the [build](https://github.com/opensearch-project/opensearch-build/pull/3305) repo. I liked the idea of having merit badges to pull requests that come in the repository.
Tested here: [Sample Job](https://github.com/owaiskazi19/OpenSearch/actions/runs/4421563834/jobs/7752499295)

Will look like this:
Screenshot 2023-03-14 at 6 07 02 PM

Open to discussion about this workflow.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by prudhvigodithi — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
@owaiskazi19 @kotwanikunal are waiting on any feedback or can we merge the change?

**Comment by prudhvigodithi** [comment]
> Late to the party! We can merge this for sure. Right now this workflow counts bot also as user. I was waiting if @prudhvigodithi can provide some insight on that how to ignore the bots (but we can incorporate that later). Also, we don't need to create the label. The workflow takes care of it.

Yes @owaiskazi19 the labels are auto created, but at this point the App users cannot be excluded with this workflow, I will take a look to improve the workflow to exclude the app users as well.
Thank you

File: .github/workflows/github-merit-badger.yml
Comment (nknize) [inline .github/workflows/github-merit-badger.yml:17]: I'm not a fan of this. I get the idea of game theory as a motivator but I think we should stick with a level playing field and avoid this hierarchical social class system. It's just not a good look and can easily turn folks away.

**Review by nknize — COMMENTED** [review]
Post merge review as I didn't see this on the first go around. 

I vote to remove this social status badging pending further discussion.

**Comment by nknize** [comment]
I like the sentiment but maybe there's a better way to go about it? Could we look into GitHub achievement badges instead? Maybe a First Time Contributor OpenSearch badge on the contributor's github profile? OpenSearch All Star Achievement for multiple contributions?

**Comment by andrross** [comment]
I like the sentiment as well, but share @nknize's concern about this introducing a hierarchical social class system. It feels a bit counter to our [Principle's for Development](https://opensearch.org/about.html#principles-for-development), specifically: 

> Respectful, approachable, and friendly.
> 
> This will be a community where you will be heard, accepted, and valued, whether you are a new or experienced user or contributor.

Every contribution should be evaluated equally on the merits, and putting a prominent badge for a first time or infrequent contributor feels like it is not a level playing field. I second the vote to remove this pending further discussion.

**Comment by owaiskazi19** [comment]
@nknize @andrross thanks for the feedback. This PR was raised to get the discussion going for the merit badge and whether we like it or not for OpenSearch. Seems like we don't. The merit badge also helps to track the metrics for the repo, Mostly, to understand how many new contributors we have and the repeat contributors.

I'll go ahead and revert this PR.

**Comment by saratvemulapalli** [comment]
Thinking about it, +1 to @andrross, @nknize feels like we are distinguishing contributors which doesn't look fair. 
I like the idea of adding github badges which are added to each profile.

**Comment by andrross** [comment]
> The merit badge also helps to track the metrics for the repo, Mostly, to understand how many new contributors we have and the repeat contributors.

Have you checked out the contributor tools in https://github.com/opensearch-project/project-tools?

**Comment by nknize** [comment]
> ...we do have a dashboard also now to track the contributions

Is that URL publicly accessible? Or will it be? Perhaps exposed on opensearch.org? 

Tracking community health is something a lot of foundations do (or at least help with). 

Another thing that could be really useful is a quarterly blog post on community health! Project-tools is a good place to start but it needs some work as it counts a lot of duplicates both at the PR (e.g., backports) and the user level (e.g., can't distinguish email duplicates). Perhaps we can work on fixing that and make that fully accessible to all maintainers. 

Maybe we also think about rotating the community health blog among all maintainers? In the future we could think about establishing a PMC and making this a responsibility of the rotating PMC chair like it's done at The ASF.

**Comment by owaiskazi19** [comment]
> Is that URL publicly accessible? Or will it be? Perhaps exposed on opensearch.org?

It's internal as of now. @bbarani can share more insight on if we are planning to make it public in the future.

**Pull request merged**
